lua/harpoon/buffer.lua

```diff
@@ -107,7 +108,7 @@ function M.setup_autocmds_and_keymaps(bufnr)
 
     vim.api.nvim_create_autocmd({ "BufLeave" }, {
         group = HarpoonGroup,
-        pattern = "__harpoon*",
+        buffer = bufnr,
```

Comment:
Buffer-local autocmds so that they get cleaned up, otherwise lots of BufLeave autocmds are created and never deleted. This is the result of `:autocmd BufLeave` after opening the harpoon menu a few times (before this fix):

```
Harpoon  BufLeave
__harpoon*<Lua 2415: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 668: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 2371: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 1500: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 2733: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 2650: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 2502: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>

... other autocmds

Harpoon  BufLeave
__harpoon*<Lua 3262: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 3029: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>
<Lua 3307: ~/projects/lua/harpoon/lua/harpoon/buffer.lua:111>

```